experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -209,11 +210,8 @@ export class FieldProxyTarget extends ProxyTarget<FieldAnchor> implements Editab
 				? (destinationField[proxyTargetSymbol] as ProxyTarget<Anchor | FieldAnchor>)
 				: this;
 
-		assert(
-			destinationFieldProxy === undefined ||
-				keyIsValidIndex(destinationIndex, this.length + 1),
-			"Index must be less than or equal to length.",
-		);
+		assertValidIndex(destinationIndex, { length: this.length });
```

Comment:
If you wanted to do what your code does, `assertValidIndex(destinationIndex, this);` would be a simpler way to do this,
but you don't want to check if the index is valid in this, you want to check if its valid in destinationFieldProxy.
Also you should have to set true for allowOnePastEnd or it should assert when moving to the end of a field.

```suggestion
		assertValidIndex(destinationIndex, destinationFieldProxy, true);
```

Reply:
It seems to complain when using `destinationFieldProxy` as an argument, saying that it does not contain a length attribute. Is there a way around this?

`Property 'length' is missing in type 'ProxyTarget<FieldAnchor | Anchor>' but required in type '{ readonly length: number; }'`

Reply:
Fixed with assert(isFieldProxyTarget(destinationFieldProxy))